pkg/puller/puller.go

```diff
@@ -104,235 +101,131 @@ func (p *Puller) manage(warmupTime time.Duration) {
+				for addr, peer := range bin {
+					peersDisconnected[addr] = peer
 				}
 			}
 
-			// EachPeerRev in this case will never return an error, since the content of the callback
-			// never returns an error. In case in the future changes are made to the callback in a
-			// way that it returns an error - the value must be checked.
-			_ = p.topology.EachPeerRev(func(peerAddr swarm.Address, po uint8) (stop, jumpToNext bool, err error) {
+			neighborhoodDepth := p.topology.NeighborhoodDepth()
+			syncRadius := p.reserveState.GetReserveState().StorageRadius
+
+			_ = p.topology.EachPeerRev(func(addr swarm.Address, po uint8) (stop, jumpToNext bool, err error) {
 				if po >= neighborhoodDepth {
-					bp := p.syncPeers[po]
-					// delete from peersDisconnected since we'd like to sync
-					// with this peer
-					delete(peersDisconnected, peerAddr.ByteString())
-
-					// within depth, sync everything
-					if _, ok := bp[peerAddr.ByteString()]; !ok {
-						// we're not syncing with this peer yet, start doing so
-						bp[peerAddr.ByteString()] = newSyncPeer(peerAddr, p.bins)
-						peersToSync = append(peersToSync, peer{addr: peerAddr, po: po})
-					} else {
-						// already syncing, recalc
-						peersToRecalc = append(peersToRecalc, peer{addr: peerAddr, po: po})
+					// add peer to sync
+					if _, ok := p.syncPeers[po][addr.ByteString()]; !ok {
+						p.syncPeers[po][addr.ByteString()] = newSyncPeer(addr, po, p.bins)
 					}
+					// remove from disconnected list as the peer is still connected
+					delete(peersDisconnected, addr.ByteString())
+				} else {
+					// outside of neighborhood, prune peer
+					p.disconnectPeer(addr.ByteString(), po)
 				}
 
-				// if peer is outside of depth, do nothing here, this
-				// will cause the peer to stay in the peersDisconnected
-				// map, leading to cancelling of its running syncing contexts.
-
 				return false, false, nil
 			}, topology.Filter{Reachable: true})
 
-			p.syncPeersMtx.Unlock()
-
-			for _, v := range peersToSync {
-				p.syncPeer(ctx, v.addr, v.po, syncRadius)
+			for addr, peer := range peersDisconnected {
+				p.disconnectPeer(addr, peer.po)
 			}
 
-			for _, v := range peersToRecalc {
-				dontSync := p.recalcPeer(ctx, v.addr, v.po, syncRadius)
-				// stopgap solution for peers that dont return the correct
-				// amount of cursors we expect
-				if dontSync {
-					peersDisconnected[v.addr.ByteString()] = v
-				}
-			}
-
-			p.syncPeersMtx.Lock()
-			for _, v := range peersDisconnected {
-				p.disconnectPeer(v.addr, v.po)
-			}
-			p.syncPeersMtx.Unlock()
+			p.recalcPeers(ctx, syncRadius)
 
 		case <-p.quit:
 			return
 		}
 	}
 }
 
-func (p *Puller) disconnectPeer(peer swarm.Address, po uint8) {
-	loggerV2 := p.logger.V(2).Register()
+// disconnectPeer cancels all existing syncing and removes the peer entry from the syncing map.
+// Must be called under lock.
+func (p *Puller) disconnectPeer(peerAddr string, po uint8) {
+	loggerV2 := p.logger
```

Comment:
This logger will log to `Debug`, not to `v=2`. Is this desirable? If so, just use `p.logger` directly. The same is in the methods below.

Reply:
this is temporary, thanks